openedx/core/djangoapps/user_authn/views/login.py

```diff
@@ -478,6 +480,32 @@ def dispatch(self, request, *args, **kwargs):
         return super(LoginSessionView, self).dispatch(request, *args, **kwargs)
 
 
+def _parse_analytics_param_for_course_id(request):
```

Comment:
**Reviewer Note**: This was moved from the shim.  I don't imagine it will affect existing calls to `login_user`, but the only way to know would be to first add metrics to look for `analytics` and `course_id`. I'm voting to skip that step, because if `course_id` is already there, it would hopefully match the one in analytics.  Let me know if you think we should do another PR first to collect this data.